features/dd-sdk-android-flags/src/main/kotlin/com/datadog/android/flags/featureflags/internal/DatadogFlagsClient.kt

```diff
@@ -195,20 +201,270 @@ internal class DatadogFlagsClient(
+                        errorCode = ErrorCode.PARSE_ERROR
+                        val typeName = FlagValueConverter.getTypeName(defaultValue::class)
+                        errorMessage = "Failed to parse value '${flag.variationValue}' as $typeName"
+
+                        featureSdkCore.internalLogger.log(
+                            InternalLogger.Level.WARN,
+                            InternalLogger.Target.MAINTAINER,
+                            { "Flag '$flagKey': $errorMessage - ${exception.message}" }
+                        )
+                    }
+                }
+
+                InternalResolution.Error(
+                    flagKey = flagKey,
+                    defaultValue = defaultValue,
+                    errorCode = errorCode,
+                    errorMessage = errorMessage
+                )
+            }
+        )
+    }
+
+    // endregion
+
+    // region Type Checking
+
+    /**
+     * Typed resolution layer that handles side effects for convenience methods.
+     *
+     * This helper method acts as a middle layer to the public API, handling:
+     * - Exposure event tracking on successful resolution (via [writeExposureEvent])
+     * - Type mismatch warning logs for developer feedback (via [InternalLogger])
+     * - Returning the typed value on success or the default value on error
+     *
+     * @param T The type of the flag value
+     * @param resolution The [InternalResolution] result from [readAndParseAssignment]
+     * @return The resolved value from [InternalResolution.Success.value] on success,
+     *         or [InternalResolution.Error.defaultValue] on error
+     */
+    private fun <T : Any> resolveTracked(resolution: InternalResolution<T>): T = when (resolution) {
+        is InternalResolution.Success -> {
+            if (flagsConfiguration.trackExposures && resolution.flag.doLog) {
+                writeExposureEvent(resolution.flagKey, resolution.flag, resolution.context)
+            }
+
+            if (flagsConfiguration.rumIntegrationEnabled && resolution.flag.doLog) {
+                logEvaluation(
+                    key = resolution.flagKey,
+                    value = resolution.value
+                )
+            }
+
+            resolution.value
+        }
+        is InternalResolution.Error -> {
+            // Only log type mismatches as warnings to help developers identify configuration issues.
+            // Other errors (FLAG_NOT_FOUND, PARSE_ERROR) are expected in normal operation.
+            if (resolution.errorCode == ErrorCode.TYPE_MISMATCH) {
+                featureSdkCore.internalLogger.log(
+                    InternalLogger.Level.WARN,
+                    InternalLogger.Target.USER,
+                    { "Flag '${resolution.flagKey}': ${resolution.errorMessage}" }
+                )
+            }
+            resolution.defaultValue
+        }
+    }
+
+    // endregion
+
+// region Helper Methods
+
+    /**
```

Comment:
nit: it seems doc comments like this and below don't add much value, so maybe we can remove them to have more focus on the code itself?